### ISSUE TITLE
feature multiplier in SAGPooling

### ISSUE THREAD

**Issue by dvirginz** [pyg-team/pytorch_geometric#885]
In SAGPooling layer, it stated that 
```multiplier (float, optional): Coefficient by which features gets
            multiplied after pooling. This can be useful for large graphs and
            when :obj:`min_score` is used. (default: :obj:`1`)
```
I see them mentioning it in the paper itself, but no heuristic was described.
How have you come across the intuition for large graphs?
What is (generally speaking) a good multiplying factor given a sparse graph with ~10k vertices?

P.S PyTorch-geometric is, without a doubt, the best PyTorch library exists.

**Comment by rusty1s** [comment]
Thank you! Sadly, I can not give you any advice here, and it might be best to contact the authors for your specific question, sorry!

**Comment by bknyaz** [comment]
`min_score` and `multiplier` is used in [our paper ](https://arxiv.org/abs/1905.02850), not the SAG one. We actually don't talk about this in our paper, because we didn't present experiments with large graphs.

We use softmax over all nodes to get attention coefficients, so for large graphs most (or all) of these values can be very small. Since we then multiply node features by these values, it will result in poor propagation of node features to the next layers and poor training overall. So we can multiply coefficients by some factor `multiplier` to fix that.

From my experience, once you have > 500 nodes in the graph, you would want to multiply by some factor, which in some my cases (unpublished) was sqrt(N), where N is either the max number of nodes in the dataset or the number of nodes in the current graph. There is no theoretical justification to use or not to use this multiplier, so it's purely heuristic and it should be tuned on your dataset. 

Hope it helps!

**Comment by dvirginz** [comment]
@bknyaz , thank you very much for your "of the record" honest answer.
I'm working with meshes (~6k vertices, ~4 edges for each vertex), and looking for a good approach to sample the relevant nodes.
Indeed currently the values drop to zero rapidly.
I'll examine what you advised and update. 
If you have any more incites to shape, I'll be happy to hear them.
Thank you!
